### ISSUE TITLE
Thanks!

### ISSUE THREAD

**Issue by coderschoolreview** [hiep04/Homework#1]
## Goal of this Assignment
The goal of this assignment was to introduce you to 2 main concepts in Machine Learning: 
- Data Visualization
- Exploratory Data Analysis. 

You learn how to query and clean data using pandas library in Python, make some plots which help to understand more about data with Seaborn library.

## Things you did well:
- You go through every questions. What a great start !
- You solve problems with comprehensive solutions and it helps me a lot on grading your solution !
- Good use of Pandas and it shows that you understand applying it very well ! 

## Things to work on:
- For question: **How many people have English 'en' as their Language of choice on the website?**
- What should do is: `ecom['Language']=='en']` already returns the Series of True or False, further step is performing summation of how many True records are there by using `sum()`: sum (ecom['Language']=='en']).
- What you actually did is: you get the data of people that has 'en' in the Language column, and you try to count the number of unique Email from them. Unintentionally, this is correct because each of person has distinct email. You still get the correct answer but your solution is not the optimum answer.

## To sum up:
- Overall, excellent work ! You are demonstrating that you are understanding the material and doing a great job of applying it. Keep it up!
- Try referencing Pandas material more to understand its functions. Pandas is very powerful !
- There are small mistakes in your assignment but if you keep practicing, this should not be a problem.
- Keep up the good work with homework assignments !

**Comment by coderschoolreview** [comment]
The goal of this assignment is for you to learn to build and evaluate a model using scikit-learn library in Python. You learn how to do Data pre-processing, splitting it into training and test sets, training your model using training sets and evaluating its performance on test sets.

Things you did well: 

- Discovering the usage of many new functions!

- Understanding how to use data to train a model using library in Python. This is a very important step!

Some minor tips:

- Question 3: You should set "annot=True" when using heatmap in order to display coefficient values: sns.heatmap(data.corr(), annot=True)

- You should take more features into your X (before splitting it), instead of just 1 feature “Avg. Session Length”.

- Question 14: You must save your prediction to y_prediction: y_prediction = lm.predict(X_test). So that you can use it later, for example when you use scatter: plt.scatter(y_test, y_prediction).

- The Mean Absolute Error(MAE), Mean Squared Error(MSE), Root Mean Squared Error(RMSE) could be used like below: metrics.mean_absolute_error(y_test, y_prediction), metrics.mean_squared_error(y_test, y_prediction), np.sqrt(metrics.mean_squared_error(y_test, y_prediction))

- In the “Residual” section, you count the different between your y_test and y_prediction, and then put it into graph to see whether it is a normal distribution. So: residuals = y_test - y_prediction, and then plot it using: sns.distplot(residuals,bins=10)

**Comment by coderschoolreview** [comment]
Hi Hiep Pham,

It seem like you are getting familiar with sentiment analysis, but still have trouble finishing your assignment.

The reason for that issue is in your " def preprocessor(text): " function, you name the emotion icon collection "emotions" but use "emoticons" later in this code: 

- text = (re.sub('[\W+', '', text.lower())+' '+' '.join(emoticons).replace('-', ''))

**Comment by coderschoolreview** [comment]
---
# Assignment 6

# Goal of this Assignment

The goal of this assignment was to introduce you to following concepts:
* Unsupervised Learning (KMeans, Hierachical Clustering)
* PCA

You learn how to use PCA for dimension reduction, KMeans, and Hierarchical Clustering. Also you learn to visualize the result of both tenichque.

# Things you did well:

* Ultilizing Pandas to read, summary and visualize data.

# To sum up:

* Good work! But you need to improve on Visualization Skill (Eg: boxplot)


# Few tips

> ![](http://i.imgur.com/lGeM6mu.png)
You're stuck at this question because you initiated PCA with 2 components but fit_transform X with 6 component (V1,... V6) and (PC1,...,PC6)
